lib/adagrams.rb

```diff
@@ -0,0 +1,157 @@
+#Adagrams
+#Danielle & Alice
+
+require 'csv'
+
+#Wave 1
+#Picks 10 letters from the letter pool
+def draw_letters
+  letters = {
+    "A" => 9,
+    "B" => 2,
+    "C" => 2,
+    "D" => 4,
+    "E" => 12,
+    "F" => 2,
+    "G" => 3,
+    "H" => 2,
+    "I" => 9,
+    "J" => 1,
+    "K" => 1,
+    "L" => 4,
+    "M" => 2,
+    "N" => 6,
+    "O" => 8,
+    "P" => 2,
+    "Q" => 1,
+    "R" => 6,
+    "S" => 4,
+    "T" => 6,
+    "U" => 4,
+    "V" => 2,
+    "W" => 2,
+    "X" => 1,
+    "Y" => 2,
+    "Z" => 1
+  }
+
+  #Creates the array of all the letters available to pick
+  letter_probability = []
+  letters.each do |key,value|
+    until value == 0
+      letter_probability << key
+      value -= 1
+    end
+  end
+
+  #Select letters and replace with 0 once they are picked
+  your_hand = []
+  count = 0
+  while count < 10 do
+    index = rand(0...letter_probability.length)
+    if letter_probability[index] != 0
+      your_hand << letter_probability [index]
+      count += 1
+      letter_probability[index] = 0
+    end
+  end
+  return your_hand
+end
+
+#Wave 2
+#Checks that the letters for the word you picked are in your hand
+def uses_available_letters?(input, letters_in_hand)
+  input = input.upcase.split("")
+  value = []
+  if input.length > letters_in_hand.length
+    return false
+  else
+    input.each do |letter|
+      input.count(letter) > letters_in_hand.count(letter) ? value << false : value << true
```

Comment:
You are doing more work than you need to! As soon as one of these checks comes up false, you can return false, right? The only case that requires you to complete the loop is if every letter matches!